account_payment_order_inter_company/tests/test_account_payment_order_inter_company.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2022 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.account_invoice_inter_company.tests.test_inter_company_invoice import (
+    TestAccountInvoiceInterCompanyBase,
+)
+
+
+class TestAccountPaymentOrderInterCompany(TestAccountInvoiceInterCompanyBase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Payment mode in company B
+        cls.payment_mode_company_b = cls.env["account.payment.mode"].create(
+            {
+                "name": "Test Credit Transfer to Suppliers",
+                "company_id": cls.company_b.id,
+                "bank_account_link": "variable",
+                "payment_method_id": cls.env.ref(
+                    "account.account_payment_method_manual_out"
+                ).id,
+            }
+        )
+        # Payment mode in company A
+        cls.payment_mode_company_a = cls.env["account.payment.mode"].create(
+            {
+                "name": "Test Direct Debit of customers",
+                "company_id": cls.company_a.id,
+                "bank_account_link": "variable",
+                "payment_method_id": cls.env.ref(
+                    "account.account_payment_method_manual_in"
+                ).id,
+            }
+        )
+
+    def test_customer_invoice_is_paid_in_company_a(self):
+        # Confirm the customer invoice in company A
+        self.invoice_company_a.with_user(self.user_company_a.id).with_context(
```

Comment:
Hi Chafique test are good but can you simplify the code between the two test can you create the follwoing method
- "_create_intercompany_customer_invoice"
- "_pay_invoice(invoice, journal, payment_mode"
So you can share code between the two test and make them perfect ?


Also we need two new case.
Creating a supplier invoice in Company B and this will create a customer invoice in company A, then pay it from
- company B supplier invoice
- company A customer invoice

Thanks